File: Jiggler/Tordial-GS-_Manifold/lib/clients/intent_client.py
# intent_client.py

import grpc
from google.protobuf import empty_pb2

import issttoft_pb2
import issttoft_pb2_grpc


class IntentClient:
    def __init__(self, host="localhost", port=50051):
        self.channel = grpc.insecure_channel(f"{host}:{port}")
        self.stub = issttoft_pb2_grpc.InferenceServiceStub(self.channel)

    # === Existing methods remain the same ===

    def stream_intent_updates(self):
        """Returns an iterator for real-time intent updates."""
        return self.stub.StreamIntentUpdates(empty_pb2.Empty())

    def close(self):
        self.channel.close()


# Example usage
if __name__ == "__main__":
    client = IntentClient(host="127.0.0.1")

    print("Listening for intent updates...\n")
    try:
        for update in client.stream_intent_updates():
            print(f"[{update.timestamp}] {update.band_id} → {update.intent_value:.4f} ({update.reason})")
    except KeyboardInterrupt:
        print("\nStopped streaming.")
    finally:
        client.close()

File: Tordial-GS-_Manifold/Tordial-GS-_Manifold/lib/clients/intent_client.dart
// lib/clients/intent_client.dart

import 'package:grpc/grpc.dart';
import '../generated/issttoft.pb.dart';
import '../generated/issttoft.pbgrpc.dart';

class IntentClient {
  late final ClientChannel _channel;
  late final InferenceServiceClient _stub;

  IntentClient({String host = 'localhost', int port = 50051}) {
    _channel = ClientChannel(
      host,
      port: port,
      options: const ChannelOptions(
        credentials: ChannelCredentials.insecure(),
        idleTimeout: Duration(minutes: 5),
      ),
    );
    _stub = InferenceServiceClient(_channel);
  }

  // === Existing methods (getAllIntentBands, getIntentBand, etc.) remain the same ===

  /// Streaming intent updates (real-time)
  Stream<IntentUpdate> streamIntentUpdates() {
    return _stub.streamIntentUpdates(Empty());
  }

  Future<void> close() async {
    await _channel.shutdown();
  }
}